src/main/java/pl/edu/agh/firecell/engine/algorithm/FirePropagator.java

```diff
@@ -84,10 +86,18 @@ private int computeBurningTimeAir(State oldState, Vector3i cellIndex) {
         return newBurningTime;
     }
 
-    private int computeBurningTimeWood(Cell oldCell, double newTemperature, int currenBurningTime) {
+    private int computeBurningTimeWood(State oldState, Vector3i cellIndex, double newTemperature, int currenBurningTime) {
+        Cell oldCell = oldState.getCell(cellIndex);
         int newBurningTime = currenBurningTime;
-        if (newTemperature > WOOD.autoIgnitionTemperature() &&
-                oldCell.burningTime() == 0 &&
+        if (oldCell.burningTime() == 0 &&
+                newTemperature > WOOD.autoIgnitionTemperature() &&
+                MAX_BURNING_TIME != 0 &&
+                oldCell.flammable()) {
+            newBurningTime++;
+        }
+        if (oldCell.burningTime() == 0 &&
```

Comment:
Why there is no else-if? These conditions can both be true, and `newBurningTime` can be incremented twice.